client/my-sites/posts/index.js

```diff
@@ -12,14 +12,25 @@ import page from 'page';
 import { navigation, siteSelection } from 'my-sites/controller';
 import postsController from './controller';
 import { makeLayout, render as clientRender } from 'controller';
+import { getSiteFragment } from 'lib/route';
 
 export default function() {
 	page(
-		'/posts/:author?/:status?/:domain?',
+		'/posts/:author(my)?/:status(published|drafts|scheduled|trashed)?/:domain?',
 		siteSelection,
 		navigation,
 		postsController.posts,
 		makeLayout,
 		clientRender
 	);
+
+	page( '/posts/*', ( { path } ) => {
+		const siteFragment = getSiteFragment( path );
+
+		if ( siteFragment ) {
+			return page.redirect( `/posts/${ siteFragment }` );
+		}
+
+		return page.redirect( '/posts' );
```

Comment:
This looks good, but I'm just wondering about two things:

- The default posts path is `/posts/my` for multi-user sites and for the All Sites view, and `/posts` for sites where the current user is the only user (which basically shows only "my" posts anyway).
It's probably more appropriate if these two redirects were to `/posts/my` instead.

- I think we always use a controller imported function for `page` callbacks.
On the other hand, I don't think I recall any other case with just one callback, used only once.
So this is not a blocker: just pointing out something that might be more consistent but I'm not really sure if it's worth it. 🙂

Reply:
Moved the redirect to `/posts/my` in d6f71ce5279fbe670be203affa65dff79fa0a83c.

- I think we always use a controller imported function for page callbacks.

I don't see much difference in doing that here, but I don't have a strong opinion either. Since this is just a simple catchall redirect, I think it makes things easier to scan.